external-crates/move/crates/move-stackless-bytecode-2/src/stackless/translate.rs

```diff
@@ -0,0 +1,843 @@
+
+        IB::LdU128(bx) => assign!(imm!(Value::U128(*(*bx))), ctxt.push_register()),
+
+        IB::CastU8 => assign!(
+            primitiveOp!(Op::CastU8, Var(ctxt.pop_register())),
+            ctxt.push_register()
+        ),
+
+        IB::CastU64 => assign!(
+            primitiveOp!(Op::CastU64, Var(ctxt.pop_register())),
+            ctxt.push_register()
+        ),
+
+        IB::CastU128 => assign!(
+            primitiveOp!(Op::CastU128, Var(ctxt.pop_register())),
+            ctxt.push_register()
+        ),
+
+        IB::LdConst(const_ref) => assign!(
+            RValue::Operand(Constant(deserialize_constant(const_ref))),
+            ctxt.push_register()
+        ),
+
+        IB::LdTrue => assign!(imm!(Value::True), ctxt.push_register()),
+
+        IB::LdFalse => assign!(imm!(Value::False), ctxt.push_register()),
+
+        IB::CopyLoc(loc) => assign!(
+            primitiveOp!(Op::CopyLoc, Var(Local(*loc as usize))),
+            ctxt.push_register()
+        ),
+
+        IB::MoveLoc(loc) => assign!(
+            primitiveOp!(Op::MoveLoc, Var(Local(*loc as usize))),
+            ctxt.push_register()
+        ),
+
+        IB::StLoc(loc) => assign!(
+            RValue::Operand(Operand::Var(ctxt.pop_register())),
+            Local(*loc as usize)
+        ),
+
+        IB::Call(bx) => {
+            let name = &bx.module.name;
+            let mut modules = ctxt.model.modules();
+            let module = modules
+                .find(|m| {
+                    m.compiled().name() == (&bx.module.name)
+                        && *m.compiled().address() == bx.module.address
+                })
+                .unwrap_or_else(|| {
+                    panic!(
+                        "Module {} with address {} not found in the model",
+                        name, bx.module.address
+                    )
+                });
+            let compiled = module.compiled();
+            let function = compiled
+                .functions
+                .get(&bx.function)
+                .unwrap_or_else(|| panic!("Function {} not found in module {}", bx.function, name));
+
+            let args = function
+                .parameters
+                .iter()
+                .map(|_| Var(ctxt.pop_register()))
+                .collect::<Vec<_>>();
+
+            let lhs = function
+                .return_
+                .iter()
+                .map(|_| ctxt.push_register())
+                .collect::<Vec<_>>();
+
+            let inst = Instruction::Assign {
+                lhs,
+                rhs: RValue::Call {
+                    function: function.name,
+                    args,
+                },
+            };
+            inst
+        }
+
+        IB::Pack(_struct_ref) => {
+            let args = _struct_ref
+                .struct_
+                .fields
+                .0
+                .iter()
+                .map(|_| Var(ctxt.pop_register()))
+                .collect::<Vec<_>>();
```

Comment:
Consider a macro or helper function for this pattern, as it is repeated a number of places.